frontend/src/app/callback/page.tsx

```diff
@@ -0,0 +1,66 @@
+"use client"
+
+import { useRouter, useSearchParams } from "next/navigation"
+import querystring from "querystring"
+import { useEffect } from "react"
+import axios, { AxiosResponse } from "axios"
+import { SPOTIFY_CLIENT_ID, SPOTIFY_SECRET } from "../../../env"
+
+const TOKEN_URL = "https://accounts.spotify.com/api/token"
+
+export async function requestSpotifyToken(
```

Comment:
I see this code is also in `src/app/Authorization.ts`. I would rather keep the (auth) domain logic in classes outside callbacks, so this only includes routing logic.